### PR TITLE
Collect requirements

### PR DESCRIPTION
**Pull request by bac**
Add makefile support for gathering the minimal set of requirements in a single directory for copying over to the charm `deps` directory.

Separate out the dependencies that are only required for building into build-requirements.txt.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3729/
Test FAILed.

File: Makefile [hunk 1]
```diff
@@ -279,12 +279,25 @@ clean-downloadcache:
...
 
+.PHONY: collect-requirements
+collect-requirements:
+	-@rm -rf $(COLLECTED_REQUIREMENTS)
+	@mkdir $(COLLECTED_REQUIREMENTS)
+	bin/pip install -d $(COLLECTED_REQUIREMENTS) --no-compile --no-index --no-dependencies --find-links $(WHEEL_CACHE) --find-links $(PYTHON_CACHE) -r requirements.txt
+	# Arch-specific wheels cannot be used.
+	-@rm -rf $(COLLECTED_REQUIREMENTS)/zope.interface*
+	-@rm -rf $(COLLECTED_REQUIREMENTS)/MarkupSafe*
+	-@cp $(CACHE)/python/zope.interface* $(COLLECTED_REQUIREMENTS)
```
Comment (jaycee) [inline Makefile]: Why are these not grabbed by pip?

**Comment by jaycee** [comment]
:+1: will QA shortly.

File: Makefile (hunk 1 above)
Comment (bac) [inline Makefile]: It is exceedingly weird and I should've documented it.

pip does grab them but without being asked turns them into architecture-specific wheels.  I see know way to make it stop.

So the ugly alternative was to manually note the ones that are trouble, delete the wheel, and replace with a tarball.

**Comment by jaycee** [comment]
QA:

Running `make dist` I get this:

```
make: *** No rule to make target `collected-requirements', needed by `dist'.  Stop.
```

I think you need to fix the dist deps to use collect-requirements or change the target to collected-requirements. Doing either worked for me.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3730/
Test FAILed.

**Comment by bac** [comment]
@jaycee Thanks for the QA.  'make dist' works now.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3731/
Test PASSed.

**Comment by frankban** [comment]
:+1:

**Comment by jaycee** [comment]
QA OK. Thanks @bac.

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**